common/common.cpp

```diff
@@ -1614,6 +1614,19 @@ std::string common_detokenize(llama_context * ctx, const std::vector<llama_token
 // Chat template utils
 //
 
+std::string common_get_builtin_chat_template(const struct llama_model * model) {
+    static const char * template_key = "tokenizer.chat_template";
+    // call with NULL buffer to get the total size of the string
+    int32_t res = llama_model_meta_val_str(model, template_key, NULL, 0);
+    if (res < 2) {
```

Comment:
What's the reason for the `2` here? `llama_model_meta_val_str` returns either -1 or the length of the string.

Reply:
Hmm yeah someone added this in the PR to fix the null terminator. I suppose that it's to make sure `model_template.size() - 1` stay positive.

In any case, we can change it to

```suggestion
    if (res < 1) {
```

Then below:

```
return std::string(model_template.data(), res);
```

What do you think?

Reply:
`model_template.size()` should always be at least 1, since the size is `res + 1`. So I don't expect the -1 to cause issues, but that should also work.

Reply:
I ended up just flipping the condition to `if (res > 0)`, it's more readable this way.